tools/src/main/java/com/nedap/archie/rmobjectvalidator/RMObjectValidator.java

```diff
@@ -96,21 +126,39 @@ private void validateAttributes(List<RMObjectValidationMessage> result, CAttribu
 
         if (emptyObservationErrors.isEmpty()) {
             result.addAll(RMMultiplicityValidation.validate(attribute, pathSoFar + "/" + rmAttributeName, attributeValue));
-            if (attribute.isSingle()) {
+            if(attribute.getChildren() == null || attribute.getChildren().isEmpty()) {
+                //no child CObjects. Run default validations only
+                String query = "/" + rmAttributeName;
+                aPathQuery = queryCache.getApathQuery(query);
+                List<RMObjectWithPath> childRmObjects = aPathQuery.findList(lookup, rmObject);
+                result.addAll(runArchetypeValidations(childRmObjects, pathSoFar + query, null));
+            }
+            else if (attribute.isSingle()) {
                 validateSingleAttribute(result, attribute, rmObject, pathSoFar);
             } else {
-                for (CObject childCObject : attribute.getChildren()) {
-                    String query = "/" + rmAttributeName + "[" + childCObject.getNodeId() + "]";
+                if(attribute.getChildren() == null || attribute.getChildren().isEmpty()) {
+                    //no child CObjects. Run default validations only
+                    String query = "/" + rmAttributeName;
                     aPathQuery = queryCache.getApathQuery(query);
                     List<RMObjectWithPath> childRmObjects = aPathQuery.findList(lookup, rmObject);
-                    result.addAll(runArchetypeValidations(childRmObjects, pathSoFar + query, childCObject));
+                    result.addAll(runArchetypeValidations(childRmObjects, pathSoFar + query, null));
```

Comment:
This block is equal to the block above and unreachable because the condition is also the same. It can probably be removed.